pkg/conf/mesos_target_conf.go

```diff
@@ -0,0 +1,225 @@
+package conf
+
+import (
+	"io/ioutil"
+	"encoding/json"
+	"github.com/golang/glog"
+	"fmt"
+	"github.com/turbonomic/turbo-go-sdk/pkg/proto"
+	"github.com/turbonomic/turbo-go-sdk/pkg/probe"
+)
+
+// Configuration Parameters to connect to a Mesos Target
+type MesosTargetConf struct {
+	// Master related - Apache or DCOS Mesos
+	Master         MesosMasterType  `json:"master"`
+	MasterIP       string		`json:"master-ip"`
+	MasterPort     string		`json:"master-port"`
+	MasterUsername string		`json:"master-user"`
+	MasterPassword string		`json:"master-pwd"`
+
+	// Scheduler or Framework related
+	Framework      	   MesosFrameworkType   `json:"framework"`
+	FrameworkIP        string		`json:"framework-ip"`
+	FrameworkPort      string		`json:"framework-port"`
+	FrameworkUser      string		`json:"framework-user"`
+	FrameworkPassword  string	        `json:"framework-pwd"`
+
+	// Action Executor related to using Layer-X
+	ActionIP       string
+	ActionPort     string
+	ActionAPI      string
+
+	// Others ?
+	SlavePort      string
+	// Login Token obtained from the Mesos Master
+	Token          string
+}
+
+// Create a new MesosTargetConf from a json file.
+// Return null config if the there are errors loading or parsing the file
+func NewMesosTargetConf(targetConfigFilePath string) (*MesosTargetConf, error) {
+	glog.Infof("[MesosClientConf] Target configuration from %s", targetConfigFilePath)
+	config, err := readConfig(targetConfigFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("[[MesosTargetConf] Error reading config "+ targetConfigFilePath + " : " , err)
+	}
+	if config == nil {
+		return nil, fmt.Errorf("[[MesosTargetConf] Null config "+ targetConfigFilePath)
+	}
+
+	// Provide framework ip for dcos mesos
+	dcosMesosTargetConf(config)
+
+	// Validate
+	ok, err := config.validate()
+	if !ok {
+		return nil, fmt.Errorf("[MesosTargetConf] Invalid config : ",  err)
+	}
+	glog.Infof("[MesosTargetConf] Mesos Target Config: %+v\n", config)
+	return config, nil
+}
+
+// Provide framework ip for dcos mesos
+func dcosMesosTargetConf(config *MesosTargetConf) {
+	// For DCOS Mesos, framework IP is not specified
+	if (config.FrameworkIP == "") {
+		config.FrameworkIP = config.MasterIP
+		config.FrameworkPort = config.MasterPort
+		config.Framework = DCOS_Marathon
+	}
+}
+
+// Create a new MesosTargetConf from a given list of AccountValues
+// Return null config if the there are errors validating the config
+func CreateMesosTargetConf(targetType string, accountValues []*proto.AccountValue) (probe.TurboTargetConf, error) {
```

Comment:
Will revisit if the targetType parameter should be string or MesosMasterType. This method would be invoked by the generic TurboProbe to create a new discovery client for a target added from the UI.